specification/testbase/resource-manager/Microsoft.TestBase/preview/2023-11-01-preview/draftpackage.json

```diff
@@ -0,0 +1,1142 @@
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./testbase.json#/parameters/TestBaseAccountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DraftPackageNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The paths with access token is returned.",
+            "schema": {
+              "$ref": "#/definitions/DraftPackageGetPathResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DraftPackageGetPath": {
+            "$ref": "./examples/DraftPackageGetPath.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.TestBase/testBaseAccounts/{testBaseAccountName}/draftPackages/{draftPackageName}/copyFromPackage": {
+      "post": {
+        "tags": [
+          "Test Base Draft Package"
+        ],
+        "operationId": "DraftPackages_CopyFromPackage",
+        "description": "Copy package file and metadata from a package to this draft package",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./testbase.json#/parameters/TestBaseAccountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DraftPackageNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CopyFromPackageOperationParameters"
+            },
+            "description": "Parameters supplied to the Test Base Draft Package CopyFromPackage operation."
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](http://example.com/codeflow?start=9&length=9)

you still need a 200 or 204 final response code, in case they use the 202 Location header polling pattern

Reply:
According to https://github.com/Azure/azure-openapi-validator/blob/main/docs/post-response-codes.md, as this operation doesn't have response payload, only 202 is allowed. The effect would be take on the draft package resource model itself. The location header will provide the operationstatus endpoint instead of this api.